crates/sqlexec/src/remote/planner.rs

```diff
@@ -185,12 +192,67 @@ impl ExtensionPlanner for DDLExtensionPlanner {
                 })))
             }
             ExtensionType::DropTables => {
-                let lp = require_downcast_lp::<DropTables>(node);
-                Ok(Some(Arc::new(DropTablesExec {
-                    catalog_version: self.catalog_version,
-                    tbl_references: lp.tbl_references.clone(),
-                    if_exists: lp.if_exists,
-                })))
+                let tmp_catalog = session_state
+                    .task_ctx()
+                    .session_config()
+                    .get_extension::<TempCatalog>()
+                    .unwrap();
+                let plan = require_downcast_lp::<DropTables>(node);
+                let mut drops = Vec::with_capacity(plan.tbl_references.len());
+                let mut temp_table_drops = Vec::with_capacity(plan.tbl_references.len());
+
+                for r in &plan.tbl_references {
+                    if tmp_catalog.contains_table(&r.name) {
+                        temp_table_drops.push(r.clone());
+                    } else if self
+                        .catalog
+                        .resolve_table(&r.database, &r.schema, &r.name)
+                        .is_some()
+                        || plan.if_exists
+                    {
+                        drops.push(r.clone());
+                    } else {
+                        return Err(DataFusionError::Plan(format!(
+                            "Table '{}' does not exist",
+                            r.name
+                        )));
+                    }
+                }
+                let exec: Arc<dyn ExecutionPlan> = match (
+                    temp_table_drops.is_empty(),
+                    drops.is_empty(),
+                ) {
+                    // both temp and remote tables
+                    (false, false) => {
+                        return Err(DataFusionError::Plan("Unable to drop temp and native tables in the same statement. Please use separate statements.".to_string()))?;
```

Comment:
this does add one new unique constraint that I can't really find a good solution for. 

You now can't drop temp and native tables in the same statement.